frontend/src/tests/components/Courses/CoursesForm.test.js

```diff
@@ -0,0 +1,114 @@
+
+        render(
+            <Router  >
+                <CoursesForm />
+            </Router>
+        );
+        await screen.findByText(/Name/);
+        await screen.findByText(/Create/);
+    });
+
+
+    test("renders correctly when passing in a Courses", async () => {
+
+        render(
+            <Router  >
+                <CoursesForm initialContents={coursesFixtures.oneCourse} />
+            </Router>
+        );
+        await screen.findByTestId(/CoursesForm-id/);
+        expect(screen.getByText(/Id/)).toBeInTheDocument();
+        expect(screen.getByTestId(/CoursesForm-id/)).toHaveValue("1");
+    });
+
+
+    test("Correct Error messsages on missing input", async () => {
+
+        render(
+            <Router  >
+                <CoursesForm />
+            </Router>
+        );
+        await screen.findByTestId("CoursesForm-submit");
+        const submitButton = screen.getByTestId("CoursesForm-submit");
+
+        fireEvent.click(submitButton);
+
+        await screen.findByText(/Name is required./);
+        expect(screen.getByText(/Name is required/)).toBeInTheDocument();
+        expect(screen.getByText(/School is required./)).toBeInTheDocument();
+        expect(screen.getByText(/Term is required./)).toBeInTheDocument();
+        expect(screen.getByText(/Start date is required./)).toBeInTheDocument();
+        expect(screen.getByText(/End date is required./)).toBeInTheDocument();
+        expect(screen.getByText(/GithubOrg is required./)).toBeInTheDocument();
+    });
+
+    test("No Error messsages on good input", async () => {
+
+        const mockSubmitAction = jest.fn();
+
+
+        render(
+            <Router  >
+                <CoursesForm submitAction={mockSubmitAction} />
+            </Router>
+        );
+        await screen.findByTestId("CoursesForm-name");
+
+        const nameField = screen.getByTestId("CoursesForm-name");
+        const schoolField = screen.getByTestId("CoursesForm-school");
+        const termField = screen.getByTestId("CoursesForm-term");
+        const startField = screen.getByTestId("CoursesForm-start");
+        const endField = screen.getByTestId("CoursesForm-end");
+        const githubOrgField = screen.getByTestId("CoursesForm-githubOrg")
+        const submitButton = screen.getByTestId("CoursesForm-submit");
+
+        fireEvent.change(nameField, { target: { value: "CMPSC 156" } });
+        fireEvent.change(schoolField, { target: { value: 'ucsb' } });
+        fireEvent.change(termField, { target: { value: 'f23' } });
+        fireEvent.change(startField, { target: { value: '2022-01-02T12:00' } });
+        fireEvent.change(endField, { target: { value: '2022-02-02T12:00' } });
+        fireEvent.change(githubOrgField, { target: { value: 'cs156-f23'}})
+        fireEvent.click(submitButton);
+
+        await waitFor(() => expect(mockSubmitAction).toHaveBeenCalled());
+
+        expect(screen.queryByText(/Start date is required./)).not.toBeInTheDocument();
```

Comment:
is there a reason you use getByText above and queryByText here?

Reply:
Using getByText when it isn't in the document will lead to an error. This test is checking if that text is not in the document.